base/io.jl

```diff
@@ -448,7 +456,50 @@ function readuntil(s::IO, t::AbstractString)
 end
 
 readline() = readline(STDIN)
-readline(s::IO) = readuntil(s, '\n')
+
+function readline(s::IO, chomp::Bool = false)
+    out = UInt8[]
+    while !eof(s)
+        c = read(s, UInt8)
+        if c == 0x0d && !eof(s) && Base.peek(s) == 0x0a
+            !chomp && push!(out, c)
+            c = read(s, UInt8)
+            !chomp && push!(out, c)
+            break
+        elseif c == 0x0a
+            !chomp && push!(out, c)
+            break
+        else
+            push!(out, c)
+        end
+    end
+
+    return String(out)
+end
+
+const linefeeds = ['\n', '\r', '\u85', '\u0B', '\u0c', '\u2028', '\u2029']
+
+function readline(s::IO, chomp = false, newlines::Vector{Char}=linefeeds)
+    out = IOBuffer()
+    while !eof(s)
+        c = read(s, Char)
+        if c in newlines
+            if c == '\r' && !eof(s) && Base.peek(s) == 0x0a
```

Comment:
Use `== '\n'`, comparison with integers is deprecated. This will be as efficient anyway.